pimd/pim_upstream.c

```diff
@@ -1865,6 +1865,8 @@ int pim_upstream_inherited_olist_decide(struct pim_instance *pim,
 					flag = PIM_OIF_FLAG_PROTO_IGMP;
 				if (PIM_IF_FLAG_TEST_PROTO_PIM(ch->flags))
 					flag |= PIM_OIF_FLAG_PROTO_PIM;
+				if (starch)
```

Comment:
starch can never be null inside of the if statement for `if (pim_upstream_evaluate_join_desired_interface)...

Let's simplify the code and start with `flag = PIM_OIF_FLAG_PROTO_STAR` and reverse the if (!ch) test case

Reply:
For SSM case, starch can be NULL. Also when (S,G) is received before (*,G) , starch will be NULL.

Reply:
Please take a few minutes and look at pim_upstream_evaluate_join_desired_interface.  The only time it returns *true* is when starch is also true.  There for this if(starch) is irrelevant.  We cannot get in here if starch is null